src/main/java/com/amazon/sqs/javamessaging/AmazonSQSExtendedClient.java

```diff
@@ -357,7 +358,22 @@ public ReceiveMessageResult receiveMessage(ReceiveMessageRequest receiveMessageR
                 String largeMessagePointer = message.getBody();
                 largeMessagePointer = largeMessagePointer.replace("com.amazon.sqs.javamessaging.MessageS3Pointer", "software.amazon.payloadoffloading.PayloadS3Pointer");
 
-                message.setBody(payloadStore.getOriginalPayload(largeMessagePointer));
+                final String originalBody;
+                try {
+                    originalBody = payloadStore.getOriginalPayload(largeMessagePointer);
+                } catch (AmazonServiceException e) {
+                    boolean isNoSuchKeyException = ((AmazonServiceException) e.getCause()).getErrorCode().equals("NoSuchKey");
```

Comment:
Why do we need casting here?

Reply:
I think that `e.getCause()` is Throwable at this point